In [2]:
import pandas as pd
import os
import re
import shutil
from rename_move_files import *

In [3]:
# cambio nombre a todos los archivos de MSI y MSS
path_comun = "/home/flori/ironhack/projects/final-project"
pathMSI = f"{path_comun}/MSIMUT_JPEG"
pathMSS = f"{path_comun}/MSS_JPEG"
MSIfiles = os.listdir(pathMSI)
MSSfiles = os.listdir(pathMSS)
print(len(MSIfiles),MSIfiles[0])
print(len(MSSfiles),MSSfiles[0])

75039 blk-DIIMCHYEWATG-TCGA-AA-3715-01Z-00-DX1.jpg
117273 blk-WAIVQLAAWVDN-TCGA-CA-6718-01Z-00-DX1.jpg


In [4]:
renameFiles(pathMSI, MSIfiles, "MSI")

'Number of files that have been renamed: 75039'

In [5]:
renameFiles(pathMSS, MSSfiles, "MSS")

'Number of files that have been renamed: 117273'

**Una vez cambiado el nombre de todos los archivos (los 75039 de MSI y los 117273 de MSS) voy a cambiarlos de carpeta**

In [6]:
# create a new folder which will contain all images. Each image will be renamed such as MSI/1 or MSS/2
new_path = "../input/biopsy_images"
if not os.path.isdir(new_path):
    os.makedirs(new_path)

In [8]:
#Declaro de nuevo estas variables para que tengan los nombres de los archivos actualizados
MSIfiles = os.listdir(pathMSI)
MSSfiles = os.listdir(pathMSS)

In [9]:
# Una vez cambiados los nombres de los archivos vamos a moverlos a la nueva carpeta creada
moveFiles(pathMSI, MSIfiles, new_path)
moveFiles(pathMSS, MSSfiles, new_path)

Number of files that have been moved: 75039
Number of files that have been moved: 117273


In [10]:
# Para ver si todo ha ido bien, voy a checkear el numero de archivos en el path antiguo y en el path nuevo
MSIfiles = os.listdir(pathMSI)
MSSfiles = os.listdir(pathMSS)
print(len(MSIfiles), len(MSSfiles))
print(len(os.listdir(new_path)))

0 0
192312
